### PR TITLE
Ali/collection show

### PR DESCRIPTION
**Pull request by mintona**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
* This PR adds the routing, controller and view necessary for a Reading Collection show page
* The route for a Reading Collection show uses a slug, not an id. The slug was added to the Reading Collections model and was also indexed. 
* Currently, this feature uses a traditional Rails MVC flow and can be modified by the FE for styling.

## Related Tickets & Documents
#21 
#22 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
n/a 

## Added tests?

- [x] yes
* Reading Collection Model test was expanded upon for the slug addition + including tests for attributes
* A spec was added to `spec/system/reading_collections/user_visits_a_collection_spec.rb` for basic feature testing.
* The full test suite is being inconsistent however, any failing test can be successfully run on its own.

- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?
* This is part of the greater Machine Collections epic that will need to be documented.

- [ ] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?
n/a
## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/controllers/reading_collections_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class ReadingCollectionsController < ApplicationController
+  def show
+    @collection = ReadingCollection.find_by(slug: params[:slug])
```
Comment (philjdelong) [inline app/controllers/reading_collections_controller.rb:3]: Slug lookin' good

File: app/models/reading_collection.rb [hunk 2]
```diff
@@ -3,5 +3,15 @@ class ReadingCollection < ApplicationRecord
   has_many :reading_collection_articles, dependent: :destroy
   has_many :articles, through: :reading_collection_articles
 
-  validates :name, :user_id, presence: true
+  validates :name, :user_id, presence: { presence: true }
+  validates :slug, presence: true, format: /\A[0-9a-z\-_]*\z/, uniqueness: { scope: :user_id }
```
Comment (philjdelong) [inline app/models/reading_collection.rb:7]: Where does the format come from? Interesting syntax

File: app/views/reading_collections/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<%= @collection.name %>
+<% @collection.articles.each do |article| %>
+  <div class="article">
+    <%= article.title %>
```
Comment (philjdelong) [inline app/views/reading_collections/show.html.erb:4]: Awesome. I know Joel will have to make changes, but this setup is perfect for what I did with _new.

File: spec/system/reading_collections/user_visits_a_collection_spec.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+require "rails_helper"
+
+RSpec.describe "Views a collection", type: :system do
+  let_it_be(:user) { create(:user) }
```
Comment (philjdelong) [inline spec/system/reading_collections/user_visits_a_collection_spec.rb:4]: This makes me think of the Beatles..but I'm glad it works for testing

**Review by philjdelong — APPROVED** [review]

File: app/controllers/reading_list_items_controller.rb [hunk 5]
```diff
@@ -3,6 +3,9 @@ def index
     @reading_list_items_index = true
     @reading_collections = ReadingCollection.where(user_id: session_current_user_id).to_json
     # Article.joins(:taggings).joins(:tags).select("articles.*").where("tags.name = 'javascript'")
+    # Article.joins(:tags).where("tags.name = 'javascript'")
+    # Article.joins(:tags).where("tags.name = 'javascript'").where("articles.created_at < ?", 2.days.ago)
+    # Article.joins(:tags).where("tags.name = 'javascript'").where("articles.created_at BETWEEN ? AND ?",Time.now-2.weeks,Time.now)
```
Comment (joel-oe-lacey) [inline app/controllers/reading_list_items_controller.rb:8]: Take it these were notes for the collections query, but these are going to be made obsolete by the new taggable process we were looking at?

File: db/migrate/20200326231610_add_slug_to_reading_collections.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddSlugToReadingCollections < ActiveRecord::Migration[5.2]
+  def change
+    add_column :reading_collections, :slug, :string
```
Comment (joel-oe-lacey) [inline db/migrate/20200326231610_add_slug_to_reading_collections.rb:3]: So to gain understanding here, this and the other slug change below just implement specific handling to generate the slug dynamically?

File: app/models/reading_collection.rb (hunk 2 above)
Comment (mintona) [reply app/models/reading_collection.rb:7]: I found it in other models in the app! Trying to be consistent.

File: app/controllers/reading_list_items_controller.rb (hunk 5 above)
Comment (mintona) [reply app/controllers/reading_list_items_controller.rb:8]: Yup! We actually moved this logic to the ReadingCollection model on Phil's branch. The logic is similar but we learned about the `acts_as_taggable` gem that really helps!

File: db/migrate/20200326231610_add_slug_to_reading_collections.rb (hunk 6 above)
Comment (mintona) [reply db/migrate/20200326231610_add_slug_to_reading_collections.rb:3]: Exactly. When a reading collection is created, it needs to have a "slug" also created. It's like an alias so you don't have to display a resources id in a route (or anywhere). It's a unique string based on the name of the reading collection. The "before_validation" creates the slug so the reading collection can be created without a human entering a slug.

**Review by joel-oe-lacey — APPROVED** [review]

**Pull request merged**